### PR TITLE
Add logic to make API audits not-applicable if no impl is found + audit rename

### PR DESCRIPTION
**Pull request by avparmar**
Add a check for an impl before proceeding with the gpt-errors-overall and deprecated-api-usage audits logic in order to make them ready for real usage.

Changed the name from deprecated-gpt-api-usage -> deprecated-api-usage.

### CODE REVIEW

File: lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js [hunk 1]
```diff
@@ -39,20 +41,29 @@ class DeprecatedApiUsage extends Audit {
    */
   static get meta() {
     return {
-      id: 'deprecated-gpt-api-usage',
+      id: 'deprecated-api-usage',
```
Comment (jburger424) [inline lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js:44]: Nit: I'd say the former name was more descriptive/accurate, and maybe just the filename should be changed instead? Core lighthouse has a generic api deprecation audit that may be confused with this from someone unfamiliar: https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/deprecations.js

**Review by jburger424 — APPROVED** [review]

File: lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js (hunk 1 above)
Comment (warrengm) [reply lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js:44]: I'm OK either way. The name is a bit of an implementation detail

**Review by warrengm — APPROVED** [review]

File: lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js (hunk 1 above)
Comment (jimper) [reply lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js:44]: FYI if you change the audit filename, you also need to change the doc filename. And we'd also need to set up a redirect for any existing links to the old doc.

Comment (avparmar) [reply lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js]: Done!

Comment (avparmar) [reply lighthouse-plugin-publisher-ads/audits/deprecated-api-usage.js]: Yeah, for now I only changed the id of the audit and its references.

**Pull request merged**